docs_src/V2TopLevelMigration.md

```diff
@@ -0,0 +1,145 @@
+    .../kv/edgex/core/2.0/core-data/
+
+The same applies for custom device and application service once they have been migrated following the guides referenced in the [Custom Device Service](custom-device-service) and [Custom Applications Service](custom-applications-service) sections below.
+
+!!! warning
+    If the Consul data is not cleared prior to running the V2 services,  the V1 configuration will remain and be taking up useful memory.  The configuration data in Consul can be cleared by deleting the `.../kv/edgex/` node with this curl command:
+
+    ````
+    curl --request DELETE http://localhost:8500/v1/kv/edgex?recurse=true`
+    ````
+
+### TOML File
+
+If you have custom configuration TOML files for any EdgeX service (core, support, device, etc.) that configuration will need to be migrated. See [V2 Migration of Common Configuration](../microservices/configuration/V2MigrationCommonConfig/) for the details on migrating configuration common to all EdgeX services.
+
+The following are where you can find the configuration migration specifics for individual core/support the services
+
+- [Core Data](../microservices/core/data/Ch-CoreData/#v2-configuration-migration-guide) 
+- [Core Metadata](../microservices/core/metadata/Ch-Metadata/#v2-configuration-migration-guide) 
+- [Core Command](../microservices/core/command/Ch-Command/#v2-configuration-migration-guide)
+- [Support Notifications](../microservices/support/notifications/Ch-AlertsNotifications/#v2-configuration-migration-guide)
+- [Support Scheduler](../microservices/support/scheduler/Ch-Scheduler/#v2-configuration-migration-guide)
+- [System Management Agent](../microservices/system-management/agent/Ch_SysMgmtAgent/#v2-configuration-migration-guide)  (DEPRECATED)
+- [Application Services](../microservices/application/V2Migration/#configuration)
+- [Device Services (common)](../microservices/device/V2Migration/#configuration)
+- Device MQTT (TBD)
+- Device Camera (TBD)
+
+### Custom Environment Overrides
+
+If you have custom [environment overrides](../microservices/configuration/CommonEnvironmentVariables/#environment-overrides) for configuration impacted by the V2 changes you will also need to migrate your overrides to use the new name or value depending on what has changed. Refer to the links above and/or below for details for migration of common and/or the service specific configuration to determine if your overrides require migrating.
+
+## Custom Device Service
+
+If you have custom Device Services they will need to be migrated to the V2 version of the Device SDK.  See [Device Service V2 Migration Guide](../microservices/device/V2Migration) for complete details.
+
+## Custom Device Profile
+
+If you have custom V1 Device Profile(s) for one of the EdgeX Device Services they will need to be migrated to the V2 version of Device Profiles.  See [Device Service V2 Migration Guide](../microservices/device/V2Migration#device-profiles) for complete details.
```

Comment:
Are our device profiles actually versioned (i.e. do they include a version)?

Reply:
> Are our device profiles actually versioned (i.e. do they include a version)?

Yes, they are define in the Swagger which is versioned.